docs: ui.button_group (#910)
deephaven/deephaven-plugins: plugins/ui/docs/components/button_group.md

# Button Group

A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button groups.

## Example

```python
from deephaven import ui

my_button_group_basic = ui.button_group(
    ui.button("Rate Now", variant="accent"),
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
)
```

## UI Recommendations

Recommendations for creating button groups:

1. The most critical action in a button group should use an accent, or negative button style, while other actions should be primary outline buttons.
2. Button groups should be left-aligned to follow content such as blocks of text, center-aligned in empty states, and right-aligned in container components like dialogs, popovers, or cards.
3. Button priority should match text alignment: for left-aligned text, the most critical button is on the left; for right- or center-aligned text, the most critical button is on the right.

> 4. Icons should be used for higher priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower priority actions.

Reviewer: ```suggestion
4. Icons should be used for higher-priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower-priority actions.
```

Consider using an [`action_group`](./action_group.md) to allow the user to select from a list of actions. 

## Content

A button group is used to handle button overflow and, thus, expects buttons as children. It switches to a vertical layout when horizontal space is limited.

```python
from deephaven import ui

my_button_group_content_space_example = ui.view(
    ui.button_group(
        ui.button("Rate Now", variant="accent", style="outline"),
        ui.button("Remind me later", variant="primary", style="outline"),
        ui.button("No, thanks", variant="primary", style="outline"),
    ),
    width=200,
    border_width="thin",
    padding="size-100",
)
```

## Orientation

Setting the `orientation` prop to "vertical" will prevent any spacing-related dynamic orientation changes.

The button group will remain in the orientation regardless of the width.

```python
from deephaven import ui

my_button_group_orientation_example = ui.button_group(
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
    ui.button("Rate Now", variant="accent"),
    orientation="vertical",
)
```

## Alignment

> By default, button groups are start-aligned to be in accordance with content, but they can be set to have a different alignment using the `alignment` prop.

Reviewer: ```suggestion
By default, button groups are start-aligned to accord with content, but they can be set to have a different alignment using the `alignment` prop.
```

        ui.button("No, thanks", variant="primary", style="outline"),
    ),
    width=200,
    border_width="thin",
    padding="size-100",
)
```

## Orientation

Setting the `orientation` prop to "vertical" will prevent any spacing-related dynamic orientation changes.

The button group will remain in the orientation regardless of the width.

```python
from deephaven import ui

my_button_group_orientation_example = ui.button_group(
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
    ui.button("Rate Now", variant="accent"),
    orientation="vertical",
)
```

## Alignment

By default, button groups are start-aligned to be in accordance with content, but they can be set to have a different alignment using the `alignment` prop.

```python
from deephaven import ui

@ui.component
def ui_button_group_alignment_examples():
    return [
        ui.button_group(
            ui.button("No, thanks", variant="secondary"),
            ui.button("Remind me later", variant="secondary"),
            ui.button("Rate Now", variant="primary"),
            orientation="vertical",

>         ),

Reviewer 2: Remove default align, don't need to show this again.

        ui.button_group(
            ui.button("No, thanks", variant="secondary"),
            ui.button("Remind me later", variant="secondary"),
            ui.button("Rate Now", variant="primary"),
            orientation="vertical",
            align="center",
        ),
        ui.button_group(
            ui.button("No, thanks", variant="secondary"),
            ui.button("Remind me later", variant="secondary"),
            ui.button("Rate Now", variant="primary"),
            orientation="vertical",
            align="end",

>         ),

Reviewer 2: Use the variant + style combination as used in all the other examples pls.

# Button Group

> A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button group.

Reviewer: ```suggestion
A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button groups.
```

# Button Group

A button group is a UI component that groups buttons with related actions together.

> Note that the button component can only be used within a button group.

Reviewer: worth formatting as a true note?

> [!NOTE]
> The button component can only be used with a button group.

Reviewer 2: Changed suggested wording to note be a note.

> A button group is a UI component that groups buttons with related actions together.
> 
> Note that the button component can only be used within a button group.

Reviewer 2: ```suggestion
A button group is a UI component that groups buttons with related actions together and will automatically handle layout overflow nicely. Only buttons can be used within button group.
```

## Example

```python

> from deephaven import ui
> 
> 
> my_button_group_basic = ui.button_group(
>     ui.button("Rate Now", variant="primary"),
>     ui.button("No, thanks", variant="secondary"),
>     ui.button("Remind me later", variant="secondary"),
> )

Reviewer 2: ```suggestion
from deephaven import ui

my_button_group_basic = ui.button_group(
    ui.button("Rate Now", variant="accent"),
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
)
```

Adjust to preferred styling.

```

## UI Recommendations

Recommendations for creating button groups:

> 1. The most critical action in a button group should use an accent, primary, or negative button style, while other actions should be secondary outline buttons.

Reviewer 2: ```suggestion
1. The most critical action in a button group should use an accent, or negative button style, while other actions should be primary outline buttons.
```

2. Button groups should be left-aligned to follow content such as blocks of text, center-aligned in empty states, and right-aligned in container components like dialogs, popovers, or cards.
3. Button priority should match text alignment: for left-aligned text, the most critical button is on the left; for right- or center-aligned text, the most critical button is on the right.
4. Icons should be used for higher priority actions if used in the button group. If the most critical action does not have an icon, avoid using icons for the other lower priority actions.

> Consider using a [`action_group`](./action_group.md) to allow the user to select from a list of actions. 

Reviewer: ```suggestion
Consider using an [`action_group`](./action_group.md) to allow the user to select from a list of actions. 
```

## Content

> Button group is used used to handle button overflow, and thus, expects buttons as children. It switches to a vertical layout when horizontal space is limited.

Reviewer: ```suggestion
A Button group is used to handle button overflow and, thus, expects buttons as children. It switches to a vertical layout when horizontal space is limited.
```

```python

> from deephaven import ui
> 
> 
> my_button_group_content_with_space_example = ui.button_group(
>     ui.button("No, thanks", variant="secondary"),
>     ui.button("Remind me later", variant="secondary"),
>     ui.button("Rate Now", variant="primary"),
> )
> 
> 
> my_button_group_content_with_less_space_example = ui.button_group(
>     ui.button("No, thanks", variant="secondary"),
>     ui.button("Remind me later", variant="secondary"),
>     ui.button("Rate Now", variant="primary"),
>     width=20,
> )

Reviewer 2: ```suggestion
from deephaven import ui

my_button_group_content_space_example = ui.view(
    ui.button_group(
        ui.button("Rate Now", variant="accent", style="outline"),
        ui.button("Remind me later", variant="primary", style="outline"),
        ui.button("No, thanks", variant="primary", style="outline"),
    ),
    width=200,
    border_width="thin",
    padding="size-100",
)
```

It's more about the parent being constrained in width, and the group auto switching orientation. This shows that better.

```

## Orientation

Setting the `orientation` prop to "vertical" will prevent any spacing-related dynamic orientation changes.

The button group will remain in the orientation regardless of the width.

```python
from deephaven import ui

my_button_group_orientation_example = ui.button_group(

>     ui.button("No, thanks", variant="secondary"),
>     ui.button("Remind me later", variant="secondary"),
>     ui.button("Rate Now", variant="primary"),

Reviewer 2: ```suggestion
    ui.button("No, thanks", variant="primary", style="outline"),
    ui.button("Remind me later", variant="primary", style="outline"),
    ui.button("Rate Now", variant="accent"),
```
